ui/app/AppLayouts/Chat/ContactsColumn/ChannelList.qml

```diff
@@ -70,19 +91,41 @@ ScrollView {
             }
         ]
 
-        function openMenu(channelIndex, muted) {
+        function openMenu(channelIndex, muted, chatType, chatName, chatId, chatIdenticon) {
             channelContextMenu.channelIndex = channelIndex
             channelContextMenu.channelMuted = muted
+            channelContextMenu.chatType = chatType
+            channelContextMenu.chatName = chatName
+            channelContextMenu.chatId = chatId
+            channelContextMenu.chatIdenticon = chatIdenticon
             channelContextMenu.popup()
         }
 
         Action {
-            //% "View Group"
-            text: qsTrId("view-group")
+            enabled: channelContextMenu.chatType !== Constants.chatTypePublic
+            text: {
+                if (channelContextMenu.chatType === Constants.chatTypeOneToOne) {
+                    return qsTrId("view-profile")
+                }
+                if (channelContextMenu.chatType === Constants.chatTypePrivateGroupChat) {
+                    return qsTrId("view-group")
+                }
+                return qsTrId("share-chat")
```

Comment:
Don't use `qsTrId` without the source comment above please. Example:
```
 //% "View Group"
text: qsTrId("view-group")
```
Without those, we get the `undefined` texts